### PR TITLE
[ACR Query Extension] Bug fix: omitting login server endpoint suffix

### PR DESCRIPTION
**Pull request by CarolineNB**
This fix prevents appending .azurecr.io to queries when users specify their ACR. 
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
 az acr query -n <registry_name> -q <query>

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
ACR Query Extension

File: src/acrquery/azext_acrquery/_params.py
Comment (huanwu) [inline src/acrquery/azext_acrquery/_params.py:11]: Can we add a test on it?

File: src/acrquery/azext_acrquery/tests/latest/test_acr_query.py
Comment (zhoxing-ms) [reply src/acrquery/azext_acrquery/tests/latest/test_acr_query.py:39]: Actually, you can try using the customized `Preparer` to avoid recording the `accessToken` for `acr login`.

> If not, I can remove it temporarily and we can replace with mocks until the feature flag is removed.

If this is only a one-time temporary need, it is also acceptable

Comment (CarolineNB) [reply src/acrquery/azext_acrquery/tests/latest/test_acr_query.py:39]: Preparer generates a new resource within a specified or new resource group correct? 

In our case, the resource would need to exist already prior to running the tests (and we would need to login to it).

Comment (CarolineNB) [reply src/acrquery/azext_acrquery/tests/latest/test_acr_query.py:39]: The test scenario we have is that we have a pre-created resources because the feature is blocked by default we need to enable it in a specific resources. Should we avoid using pre-created resources for testing?

Comment (zhoxing-ms) [reply src/acrquery/azext_acrquery/tests/latest/test_acr_query.py:39]: @CarolineNB Yes, please use the dynamically created resources in the test instead of pre-created resources to avoid the problem of dependent resources being deleted will cause the test to be failed

**Comment by CarolineNB** [comment]
Hey @zhoxing-ms, tests have been added. Let me know if there's anything else needed before merging. Thanks!

File: src/acrquery/azext_acrquery/_params.py
Comment (huanwu) [inline src/acrquery/azext_acrquery/_params.py:12]: Question, do we need update history.rst?

File: src/acrquery/HISTORY.rst
Comment (jaysterp) [inline src/acrquery/HISTORY.rst:5]: For bug fix, please bump the patch version number. 1.0.0 -> 1.0.1

**Review by huanwu — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
Please take a look at this comment https://github.com/Azure/azure-cli-extensions/pull/6606#issuecomment-1668670982. If you want to release a new extension version for this PR, please also modify the `setup.py`

**Comment by CarolineNB** [comment]
Added, thanks @zhoxing-ms!

**Comment by CarolineNB** [comment]
Hey @zhoxing-ms, is there anything else that is needed for this PR?

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@jaysterp Could you please review this PR again?

**Pull request merged**
